Alerting/integrations error feedback handle null dates in response

soniaAguilarPeiron opened the pull request:
**What this PR does / why we need it**:

This PR updates fake response and handles null date in ReceiversTable, according to latest changes in the alert manager. See gist : 

https://gist.github.com/santihernandezc/dfce8a61e7e67e46211be715700c56ed/revisions

**Special notes for your reviewer**:

This PR is going to be merged to the `main-branch-for-receivers-integrations-error-feedback`

soniaAguilarPeiron committed:
Alerting: Receivers integrations error feedback: WIP - Add notifications error at the top right on contact points view (#52390)

* Add interfaces for contact point errors

* [WIP] Create fake response for the new service to get contact point errors

* [WIP] Create action an reducer for the new service to get contact point errors

* Fetch fetchContactPointStates in Contact Points tab every 20s and when AM changes

* [WIP] Use store to get error count

* Show number of integrations errors at the contact points main view

* Add warning icon and refactor styles using getStyles

* Change lastNotify type to string instead of DateTime

* Use Stack component from experimental library when it is possible

soniaAguilarPeiron committed:
Alerting: Add receivers error feedback in contact point list (#52524)

* Refactor types for contact points state

* Add health column in ReceiversTable in case error state is available for this AM

* Create method for converting contact points state DTO to the FE type used in Redux store

* Update types

* Fix indexOf criteria getting integration type

* Change type name to integrationType name

* Change new components to be named functions to follow the FE style-guide

* Fix typos

Co-authored-by: Konrad Lalik <konrad.lalik@grafana.com>

* Decouple ReceiversTable from Redux state

* Create private useContactPointsState hook to simplify code in ReceiversTable component

* Add tests for getIntegrationType and refactor the method to validate the name

* Add tests for contactPointsStateDtoToModel method

* Remove unnecessary check

* Use Badge compoment for health status in contact point list

* Create new method parseIntegrationName to simplify getting types and index from integration name

Co-authored-by: Konrad Lalik <konrad.lalik@grafana.com>

soniaAguilarPeiron committed:
Alerting: Show integrations error feedback when expanding contact point in list (#52920)

* Use DynamicTable for rendering list of contact points and make them expandable if error status is available

* Render expanded content for contact points integrations

* Style and format last notify column

* Add send resolve column to the integration details

* Fix receiver id for DynamicTable row

* Update clock icon in integration state

* Fix tests

* Add PR review sugestions

soniaAguilarPeiron committed:
Update fake response with lastNotify ISO8601 formatted, to be aligned with latest BE changes

soniaAguilarPeiron committed:
Update LastNotify in ReceiversTable component to handle null date

sakjur committed:
Org: Fix service initialization (#55460)

joshhunt committed:
Chore: Type Rudderstack and AppInsights window APIs (#55162)

* Chore: Type Rudderstack and AppInsights window APIs

* remove rudder sdk to dev deps

Jguer committed:
Auth: Add SAML common resolutions to cookie issues (#55395)

* Auth: Add SAML cookie specifics

* Auth: remove legacy message

* HTTPS

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 9.1.6 (#55481)

xlson committed:
Chore: update latest.json to 9.1.6 (#55486)

ArturWierzbicki committed:
SearchV2: add tracing to background jobs (#55250)

* searchv2: add tracing to background jobs

* searchv2: lint

* searchv2: lint

* searchv2: fix context passing

* searchv2: add init org index span

* searchv2: add traceid to logs

* searchv2: add db count to logs

renovate[bot] committed:
Update dependency @babel/plugin-transform-runtime to v7.19.1 (#55289)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

alexweav committed:
Alerting: Extract default message embed into named constant (#55424)

* Resolve conflicts

* Combine multiple const definitions

renovate[bot] committed:
Update dependency @daybrush/utils to v1.10.0 (#55492)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

FZambia committed:
Live: Migrate to centrifuge-js v3 (new API and client protocol) (#51977)

Jguer committed:
Auth: Add development manual override (#55496)

* Auth: Add development manual override

* rename to disable_sync_lock

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 8.5.13 (#55502)

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 9.0.9 (#55506)

marefr committed:
Logging: Introduce API for contextual logging (#55198)

Introduces a FromContext method on the log.Logger interface that 
allows contextual key/value pairs to be attached, e.g. per request, 
so that any logger using this API will automatically get the per request 
context attached. The proposal makes the traceID available for 
contextual logger , if available, and would allow logs originating from 
a certain HTTP request to be correlated with traceID.
In addition, when tracing not enabled, skip adding
traceID=00000000000000000000000000000000
to logs.

owensmallwood committed:
PublicDashboards: Tag is rerendered when dashboard meta changes in state (#55414)

Can update dashboard meta state using dashboard events.

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 9.0.9 (#55511)

torkelo committed:
Refactoring and simplifying TopSearchBar (#55341)

zserge committed:
Chore: Switch over to team.Service instead of sqlstore (#55497)

* switch to using team service

* trying to fix tests

* more tests to fix

* add missing teamtest package

kaydelaney committed:
PanelHeader: Add deadzone to panel header click/drag detection (#55490)

IevaVasiljeva committed:
resolve merge conflicts (#55503)

yuri-tceretian committed:
Alerting: Refactor WeCom notifier to use encoding/json to parse settings instead of simplejson (#55423)

VikaCep commented on public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx:
```diff
@@ -135,15 +135,22 @@ interface NotifiersTableProps {
-      <div>{dateTimeFormat(lastNotifyDuration, { format: 'YYYY-MM-DD HH:mm:ss' })}</div>
-    </Stack>
-  );
+  const lastNotifyDate = lastNotify;
```
What's the reason for adding this variable? can't you use `lastNotify` directly? Or rename the parameter.

soniaAguilarPeiron replied:
You're completelly right!!!

ArturWierzbicki committed:
Search: improve logging in case of failed dashboard loading phase (#55523)

ArturWierzbicki committed:
Search: Add search index configuration options (#55525)

* Search: externalize config

* Search: update config descriptions

* Search: fix value

* Search: fix

* update

* Search: revert config values

* Search: rename copy/paste

* Search: fix tests

ryantxu committed:
Search: Investigate frontend search options (again) (#55526)

Commit:
Status History Panel: Show X-Axis Value in Tooltip (#54563)

Co-authored-by: gitstart <gitstart@users.noreply.github.com>
Co-authored-by: gitstart <gitstart@gitstart.com>
Co-authored-by: Nitesh Singh <nitesh.singh@gitstart.dev>
Co-authored-by: Rubens Rafael <70234898+RubensRafael@users.noreply.github.com>
Co-authored-by: Rafael Toledo <87545086+Toledodev@users.noreply.github.com>
Co-authored-by: Matheus Benini Ferreira <88898100+MatheusBeniniF@users.noreply.github.com>
Co-authored-by: Matheus Benini <matheus_benini@hotmail.com>
Co-authored-by: Júlio Piubello da Silva Cabral <julio.piubello@gitstart.dev>
Co-authored-by: juliopiubello <juliopiubellow@gmail.com>

torkelo committed:
GrafanaRoute: Use React.Lazy instead of react-loadable and improve error handling (#55339)

* Things are working

* Add new GrafanaRoute tests

* removed old file

* Remove from package.json

Commit:
Dashboard: Single right-aligned y axis loses gridlines (#54206)

Co-authored-by: gitstart <gitstart@gitstart.com>
Co-authored-by: gitstart <gitstart@users.noreply.github.com>
Co-authored-by: Nitesh Singh <nitesh.singh@gitstart.dev>
Co-authored-by: Murilo Amaral <87545137+MuriloAmarals@users.noreply.github.com>
Co-authored-by: Matheus Benini Ferreira <88898100+MatheusBeniniF@users.noreply.github.com>
Co-authored-by: Rafael Toledo <87545086+Toledodev@users.noreply.github.com>
Co-authored-by: Rubens Rafael <70234898+RubensRafael@users.noreply.github.com>
Co-authored-by: Júlio Piubello da Silva Cabral <julio.piubello@gitstart.dev>
Co-authored-by: Thiago Nascimbeni <tnascimbeni@gmail.com>

Co-authored-by: gitstart <gitstart@users.noreply.github.com>
Co-authored-by: gitstart <gitstart@gitstart.com>
Co-authored-by: Nitesh Singh <nitesh.singh@gitstart.dev>
Co-authored-by: Murilo Amaral <87545137+MuriloAmarals@users.noreply.github.com>
Co-authored-by: Matheus Benini Ferreira <88898100+MatheusBeniniF@users.noreply.github.com>
Co-authored-by: Rafael Toledo <87545086+Toledodev@users.noreply.github.com>
Co-authored-by: Rubens Rafael <70234898+RubensRafael@users.noreply.github.com>
Co-authored-by: Júlio Piubello da Silva Cabral <julio.piubello@gitstart.dev>
Co-authored-by: Thiago Nascimbeni <tnascimbeni@gmail.com>

julienduchesne committed:
Drone: Always have `image_pull_secrets` (#55530)

Having it doesn't prevent pulling any images, so it's easier if it's everywhere

Commit:
Elasticsearch: Rename public/app/plugins/datasource/elasticsearch/query_builder.ts to follow convention (#55413)

Commit:
Elasticsearch: rename public/app/plugins/source/elasticsearch/query_def.ts to follow convention (#55415)

sunker committed:
move describe log groups to resource api (#55485)

grobinson-grafana committed:
Alerting: Fix mathexp.NoData cannot be reduced (#55347)

This commit fixes a bug where queries from datasources such as InfluxDB that returned no data would not create a DatasourceNoData alert, but instead an error "can only reduce type series, got type noData".

grobinson-grafana committed:
Alerting: Change screenshots to use components (#55156)

* Alerting: Change screenshots to use components

This commit changes screenshots to use a number of components instead of a set of functional wrappers.

It moves the uploading of screenshots from the screenshot package to the image package so we can re-use the same code for both uploading screenshots and server-side images; SingleFlight from the screenshot package to the image package so we can use it for both taking and uploading the screenshot, where as before it was used just for taking the screenshot; and it also removes the use of a cache because we know that screenshots can be taken at most once per tick of the scheduler.

gabor committed:
elastic: devenv: add nested objects to data generator (#55482)

torkelo committed:
NavBarMenu: Section as links and design tweak  (#55538)

* scaffold new component + remove storing of expanded state

* some padding fixes

* simplify!

* move browse back to being a child of dashboards

* behaviour working

* improve child matcher to look recursively

* increase NavBarMenu zIndex to ensure it overlays explore drawer

* some renaming

* fix unit test

* make dashboards a top level item again and make chevrons their own buttons

* remove active background state

* Finished tweaks

* remove theme change

* Remove exit animation

* align button centrally + fix empty message alignment

* only show the empty message if there are no children

* ensure overflowing menu items truncate correctly

Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

sakjur committed:
Cleanup: Add traces to cleanup jobs (#55465)

wbrowne committed:
Plugins: Add secure JSON fields to plugin setting DTO (#55313)

* add secure JSON fields to plugin setting DTO

* add nil pointer fix

* adding secureJsonFields to the plugin meta.

Co-authored-by: Marcus Andersson <marcus.andersson@grafana.com>
